File: 승준/20230710/drawing_line_2170.js
const readline = require("readline");

let n;
const lines = [];

readline
  .createInterface(process.stdin, process.stdout)
  .on("line", (line) => {
    if (!n) {
      n = parseInt(line);
    } else {
      const [x, y] = line.split(" ").map(Number);
      lines.push([x, y]);
    }
  })
  .on("close", () => {
    lines.sort((a, b) => a[0] - b[0]);

    let total = 0;
    let start = lines[0][0];
    let end = lines[0][1];

    for (let i = 1; i < n; i++) {
      if (lines[i][0] > end) {
        total += end - start;
        start = lines[i][0];
        end = lines[i][1];
      } else {
        end = Math.max(end, lines[i][1]);
      }
    }
    total += end - start;
    console.log(total);
    process.exit();
  });

File: 승준/20230710/drawing_line_2170.py
import sys
input = sys.stdin.readline

n = int(input())
        
lines = []

for _ in range(n):
    x, y = map(int, input().split())
    lines.append((x, y))
lines.sort()

total_length = 0
start = lines[0][0]
end = lines[0][1]

for i in range(1, n):
    if lines[i][0] > end:
        total_length += end - start
        start = lines[i][0]
        end = lines[i][1]
    else:
        end = max(end, lines[i][1])
        
total_length += end - start
print(total_length)